lib/ocpp/v201/charge_point.cpp

```diff
@@ -470,27 +470,76 @@ void ChargePoint::init_websocket() {
     this->websocket->register_connected_callback([this](const int security_profile) {
         this->message_queue->resume();
         this->websocket_connection_status = WebsocketConnectionStatusEnum::Connected;
-    });
 
-    this->websocket->register_closed_callback(
-        [this, connection_options, configuration_slot](const websocketpp::close::status::value reason) {
-            EVLOG_warning << "Failed to connect to NetworkConfigurationPriority: "
-                          << this->network_configuration_priority + 1
-                          << " which is configurationSlot: " << configuration_slot;
-            this->websocket_connection_status = WebsocketConnectionStatusEnum::Disconnected;
-            this->message_queue->pause();
-
-            if (!this->disable_automatic_websocket_reconnects) {
-                this->websocket_timer.timeout(
-                    [this, reason]() {
-                        if (reason != websocketpp::close::status::service_restart) {
-                            this->next_network_configuration_priority();
+        if (this->registration_status == RegistrationStatusEnum::Accepted) {
+            // handle offline threshold
+            //  Get the current time point using steady_clock
+            std::chrono::steady_clock::duration offline_duration = std::chrono::steady_clock::now() - time_disconnected;
+
+            // B04.FR.01
+            // If offline period exceeds offline threshold then send the status notification for all connectors
+            if (offline_duration > (std::chrono::seconds)this->device_model->get_value<int>(
```

Comment:
nit: Prefer not to cast but use

`std::chrono::seconds(this->device_model->get_value<int>(
                                       ControllerComponentVariables::OfflineThreshold))`

instead